packages/backend-defaults/src/entrypoints/urlReader/lib/GerritUrlReader.ts

```diff
@@ -217,4 +186,46 @@ export class GerritUrlReader implements UrlReaderService {
       stripFirstDirectory: false,
     });
   }
+
+  private async getRevisionForUrl(
+    url: string,
+    options?: UrlReaderServiceReadTreeOptions,
+  ): Promise<string> {
+    const { ref, refType } = parseGitilesUrlRef(this.integration.config, url);
+    // The url points to a static revision.
+    if (refType === 'sha') {
+      if (options?.etag === ref) {
+        throw new NotModifiedError();
+      }
+      return ref;
+    }
+
+    const apiUrl = getGerritBranchApiUrl(this.integration.config, url);
+    let response: Response;
+    try {
+      response = await fetch(apiUrl, {
+        method: 'GET',
+        ...getGerritRequestOptions(this.integration.config),
+      });
+    } catch (e) {
+      throw new Error(`Unable to read branch state ${url}, ${e}`);
+    }
+
+    if (response.status === 404) {
+      throw new NotFoundError(`Not found: ${url}`);
+    }
+
+    if (!response.ok) {
+      throw new Error(
+        `${url} could not be read as ${apiUrl}, ${response.status} ${response.statusText}`,
+      );
+    }
```

Comment:
so that more errors can be mapped properly by the error handler:

```suggestion
    if (!response.ok) {
      throw await ResponseError.fromResponse(response);
    }
```